### PR TITLE
`replacePkgName` handle cases of nested slices and nested maps

### PR DESCRIPTION
**Pull request by A-Hilaly**
Issue N/A

Description of changes:
- `replacePkgName` handle cases where `goType` is a nested slice or a nested map
- detach `replacePkgName` from `model.CRD` and change it signature
- Add unit tests for `replacePkgName`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/model/crd.go [hunk 1]
```diff
@@ -2294,19 +2294,25 @@ func (r *CRD) goCodeSetOutputForScalar(
...
+	apiPkgName string,
 	replacePkgAlias string,
 	keepPointer bool,
 ) string {
 	memberType := subject
+	sliceDepth := 0 // Depth of the slice type
 	isSliceType := strings.HasPrefix(memberType, "[]")
 	if isSliceType {
-		memberType = memberType[2:]
+		sliceDepth = strings.LastIndex(subject, "[]")/2 + 1
```
Comment (jaypipes) [inline pkg/model/crd.go:2307]: heh, OK :)

File: pkg/model/crd_test.go [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+		},
+		{ // don't keep pointer
+			"*ecr.Repository",
+			"ecr",
+			"svcsdk",
+			false,
+			"svcsdk.Repository",
+		},
+		{ // non sdk type
+			"*anotherpackage.Repository",
```
Comment (jaypipes) [inline pkg/model/crd_test.go:33]: let's use a specific real-world example here, yeah? How about `*time.Time`?

File: pkg/model/crd_test.go [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+			"ecr",
+			"svcsdk",
+			true,
+			"*anotherpackage.Repository",
+		},
+		{ // map type
```
Comment (jaypipes) [inline pkg/model/crd_test.go:39]: Do we happen to have a specific real-world example from one of our APIs?

File: pkg/model/crd_test.go [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+			"map[string]*ecr.Repository",
+			"ecr",
+			"svcsdk",
+			true,
+			"map[string]*svcsdk.Repository",
+		},
+		{ // nested map type
```
Comment (jaypipes) [inline pkg/model/crd_test.go:46]: Do we happen to have a specific real-world example from one of our APIs?

File: pkg/model/crd_test.go [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+			"map[string]map[string]*svcsdk.Repository",
+		},
+		{ // slice type
+			"[]ecr.Repository",
+			"ecr",
+			"svcsdk",
+			true,
+			"[]svcsdk.Repository",
+		},
+		{ // nested slices type
```
Comment (jaypipes) [inline pkg/model/crd_test.go:60]: Do we happen to have a specific real-world example from one of our APIs?

**Review by jaypipes — CHANGES_REQUESTED** [review]
this is great, Amine. Just would love to put some real-world examples into the unit test scenarios to make it clearer which specific types this code is fixing up/covering.

File: pkg/model/crd_test.go (hunk 5 above)
Comment (A-Hilaly) [reply pkg/model/crd_test.go:60]: Done! with an example from codedeploy

File: pkg/model/crd_test.go (hunk 3 above)
Comment (A-Hilaly) [reply pkg/model/crd_test.go:39]: Yes, `map[string]*ecr.Repository` is a real example in the `ecr` service

**Review by jaypipes — APPROVED** [review]
:+1:

File: pkg/model/crd_test.go (hunk 4 above)
Comment (A-Hilaly) [reply pkg/model/crd_test.go:46]: I couldn't find a real example within the services i generated.. I'm using  `map[string]map[string]uint8` for now.. WDYT @jaypipes ?

File: pkg/model/crd_test.go (hunk 2 above)
Comment (A-Hilaly) [reply pkg/model/crd_test.go]: Yes, it's a good example!

**Pull request merged**